lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -478,7 +480,7 @@ def headers(options = {})
           "X-Stripe-Client-User-Agent" => user_agent,
           "X-Stripe-Client-User-Metadata" => {:ip => options[:ip]}.to_json
         }
-        headers.merge!("Idempotency-Key" => idempotency_key) if idempotency_key
+        headers.merge!("Idempotency-Key" => idempotency_key) if idempotency_key && @enable_idempotency
```

Comment:
Can't we just delete the key from the options for void requests instead of introducing this additional piece of state?